fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/highlight/LuceneScaleTest.java

```diff
@@ -372,24 +436,60 @@ private class DataModel {
         int maxDocId = 0;
         Random random = new Random();
         private boolean continuing;
-        private final List<String> searchWords = new ArrayList<>(SEARCH_WORD_COUNT);
+        private final List<String> searchWords;
         private static final int SEARCH_WORD_COUNT = 20;
         private int lastSearchWordsUpdate = -10000;
 
+        private DataModel() {
+            if (!Config.COMMANDS_TO_RUN.contains(Command.Search) &&
+                !Config.COMMANDS_TO_RUN.contains(Command.Insert)) {
+                // this is set to immutable so that if anything tries to change it the test will fail, rather than
+                // quietly doing weird things
+                searchWords = List.of();
+            } else {
+                searchWords = new ArrayList<>(SEARCH_WORD_COUNT);
+            }
+        }
+
 
         void prep() {
+            OnlineIndexer.Builder indexBuilder = null;
             try (FDBRecordContext context = openContext()) {
+                final FDBRecordStore store = openStore(context);
                 maxDocId = context.asyncToSync(FDBStoreTimer.Waits.WAIT_LOAD_SYSTEM_KEY,
-                        openStore(context).scanRecords(TupleRange.ALL, null, ScanProperties.FORWARD_SCAN)
-                                .getCount());
+                        store.scanRecords(TupleRange.ALL, null, ScanProperties.FORWARD_SCAN).getCount());
                 continuing = maxDocId > 0;
+                if (Config.DISABLE_INDEX) {
+                    logger.info("Disabling index");
+                    store.markIndexDisabled(INDEX.getName());
+                } else {
+                    if (!store.isIndexReadable(INDEX.getName())) {
+                        indexBuilder = OnlineIndexer.newBuilder()
+                                .setRecordStoreBuilder(store.asBuilder())
+                                .setTargetIndexesByName(List.of(INDEX.getName()));
+                    }
+                }
+                context.commit();
+            }
+            if (indexBuilder != null) {
+                logger.info("Building index");
```

Comment:
Is this happenning synchronously, which is why it should not be done inside the transaciton?

Reply:
When using this for large stores, as intended, this would take way too long to happen in a single transaction.